### PR TITLE
Add additional broadcast validation tests for Fulu/PeerDAS 

### LINKED ISSUE #6855: Add additional broadcast validation tests for Fulu/PeerDAS
The tests in [broadcast_validation_tests.rs](https://github.com/sigp/lighthouse/blob/b7da07573c69b1d1c0eafaa6d656656bba42031f/beacon_node/http_api/tests/broadcast_validation_tests.rs) covers many blob scenarios, and currently tests the Electra fork.

From the Fulu fork, data columns instead of blobs will be transmitted on gossip, and we need some additional test cases to cover this - at least an equivalent test for each gossip blob scenario.

### PR DESCRIPTION
**Pull request by eserilev**
## Issue Addressed

Closes #6855

## Proposed Changes

Add PeerDAS broadcast validation tests and fix a small bug where `sampling_columns_indices` is none (indicating that we've already sampled the necessary columns) and `process_gossip_data_columns` gets called

## Additional Info

If theres a better way to format these tests (maybe have CI feed in a list of forks to the relevant tests?) I can do that instead

### CODE REVIEW

File: beacon_node/http_api/tests/broadcast_validation_tests.rs [hunk 1]
```diff
@@ -1573,14 +1613,29 @@ pub async fn blobs_seen_on_gossip_without_block() {
...
+    let deneb_enabled_forks = ForkName::list_all()
+        .into_iter()
+        .filter(|f| f.deneb_enabled())
+        .collect::<Vec<_>>();
+
+    let futures = deneb_enabled_forks
+        .into_iter()
+        .map(blobs_or_columns_seen_on_gossip_without_block_and_no_http_blobs_or_columns);
+
+    join_all(futures).await;
```
Comment (michaelsproul) [inline beacon_node/http_api/tests/broadcast_validation_tests.rs]: I agree this pattern feels a bit repetitive.

We could do what we do in the other tests like `beacon_chain`, `op_pool`, `network`, which is use `test_spec` to initialise the `ChainSpec` from the `FORK_NAME` environment variable. This would require us to add some makefile & CI support for the HTTP tests, and exclude them from `make test-release`.

**Review by michaelsproul — DISMISSED** [review]

File: beacon_node/http_api/tests/broadcast_validation_tests.rs (hunk 1 above)
Comment (eserilev) [reply beacon_node/http_api/tests/broadcast_validation_tests.rs]: yep sounds good!

File: .github/workflows/test-suite.yml [hunk 2]
```diff
@@ -159,6 +159,28 @@ jobs:
     - name: Show cache stats
       if: env.SELF_HOSTED_RUNNERS == 'true'
       run: sccache --show-stats
+  http-api-tests:
```
Comment (jimmygchen) [inline .github/workflows/test-suite.yml:162]: CI is failing because this job hasn't been added to `test-suite-success`:

https://github.com/sigp/lighthouse/blob/e0c1f27e1303ba0b4f86585241e7f88ccb4a9bbb/.github/workflows/test-suite.yml#L469-L481

File: Makefile [hunk 3]
```diff
@@ -34,6 +34,9 @@ PROFILE ?= release
 # they run for different forks.
 FORKS=phase0 altair bellatrix capella deneb electra fulu
 
+# List of all recent hard forks. This list is used to set env variables for http_api tests
+RECENT_FORKS=deneb electra fulu
```
Comment (jimmygchen) [inline Makefile]: we can probably delete deneb now

Comment (eserilev) [reply Makefile]: deleted deneb

File: beacon_node/http_api/tests/broadcast_validation_tests.rs [hunk 4]
```diff
@@ -49,7 +50,8 @@ pub async fn gossip_invalid() {
     // `validator_count // 32`.
     let validator_count = 64;
     let num_initial: u64 = 31;
-    let tester = InteractiveTester::<E>::new(None, validator_count).await;
+    let spec = test_spec::<E>();
```
Comment (jimmygchen) [inline beacon_node/http_api/tests/broadcast_validation_tests.rs]: Why is this necessary?
Looks like it defaults to `test_spec` here?
https://github.com/sigp/lighthouse/blob/11bcccb353c073ab6d23690d0fd3a3b8940d8e95/beacon_node/beacon_chain/src/test_utils.rs#L419

**Review by jimmygchen — CHANGES_REQUESTED** [review]
This is great, i didn't realise the tests can be reused without needing any modification 😄 I think we could probably remove `test_spec()` in the tests that uses `InteractiveTester::<E>::new(None,...`  as it already defaults to `test_spec`.

File: beacon_node/http_api/tests/broadcast_validation_tests.rs (hunk 4 above)
Comment (jimmygchen) [reply beacon_node/http_api/tests/broadcast_validation_tests.rs]: @eserilev I'm going to address this minor comment if you don't mind - would be great to get this merged given it's in a good shape and all passing.

Comment (eserilev) [reply beacon_node/http_api/tests/broadcast_validation_tests.rs]: sounds good, thank you!

File: beacon_node/http_api/tests/broadcast_validation_tests.rs [hunk 5]
```diff
@@ -1455,14 +1431,13 @@ pub async fn block_seen_on_gossip_without_blobs_or_columns() {
...
     let validation_level: Option<BroadcastValidation> = Some(BroadcastValidation::Gossip);
 
     // Validator count needs to be at least 32 or proposer boost gets set to 0 when computing
     // `validator_count // 32`.
     let validator_count = 64;
     let num_initial: u64 = 31;
-    let tester = InteractiveTester::<E>::new(Some(spec.clone()), validator_count).await;
+    let tester = InteractiveTester::<E>::new(None, validator_count).await;
     let state = tester.harness.get_current_state();
     let fork_name = state.fork_name(&spec).unwrap();
```
Comment (jimmygchen) [inline beacon_node/http_api/tests/broadcast_validation_tests.rs]: oh you'll need to `&tester.harness.spec` here to compile

File: beacon_node/http_api/tests/broadcast_validation_tests.rs [hunk 6]
```diff
@@ -1388,14 +1365,13 @@ pub async fn blinded_equivocation_full_pass() {
...
 pub async fn block_seen_on_gossip_without_blobs_or_columns() {
-    let spec = test_spec::<E>();
     let validation_level: Option<BroadcastValidation> = Some(BroadcastValidation::Gossip);
 
     // Validator count needs to be at least 32 or proposer boost gets set to 0 when computing
     // `validator_count // 32`.
     let validator_count = 64;
     let num_initial: u64 = 31;
-    let tester = InteractiveTester::<E>::new(Some(spec.clone()), validator_count).await;
+    let tester = InteractiveTester::<E>::new(None, validator_count).await;
```
Comment (jimmygchen) [inline beacon_node/http_api/tests/broadcast_validation_tests.rs:1374]: same here

File: beacon_node/http_api/tests/broadcast_validation_tests.rs [hunk 7]
```diff
@@ -1388,14 +1365,13 @@ pub async fn blinded_equivocation_full_pass() {
...
     let state = tester.harness.get_current_state();
     let fork_name = state.fork_name(&spec).unwrap();
```
Comment (jimmygchen) [inline beacon_node/http_api/tests/broadcast_validation_tests.rs]:
```suggestion
    let fork_name = state.fork_name(&tester.harness.spec).unwrap();
```

File: beacon_node/http_api/tests/broadcast_validation_tests.rs (hunk 5 above)
Comment (jimmygchen) [inline beacon_node/http_api/tests/broadcast_validation_tests.rs]:
```suggestion
    let fork_name = state.fork_name(&tester.harness.spec).unwrap();
```

File: beacon_node/http_api/tests/broadcast_validation_tests.rs [hunk 8]
```diff
@@ -1611,14 +1586,13 @@ pub async fn blobs_or_columns_seen_on_gossip_without_block() {
...
```
Comment (jimmygchen) [inline beacon_node/http_api/tests/broadcast_validation_tests.rs]:
```suggestion
    let fork_name = state.fork_name(&tester.harness.spec).unwrap();
```

File: beacon_node/http_api/tests/broadcast_validation_tests.rs [hunk 9]
```diff
@@ -1691,8 +1665,7 @@ async fn slashable_blobs_or_columns_seen_on_gossip_cause_failure() {
     // `validator_count // 32`.
     let validator_count = 64;
     let num_initial: u64 = 31;
-    let spec = test_spec::<E>();
-    let tester = InteractiveTester::<E>::new(Some(spec.clone()), validator_count).await;
+    let tester = InteractiveTester::<E>::new(None, validator_count).await;
     let state = tester.harness.get_current_state();
     let fork_name = state.fork_name(&spec).unwrap();
```
Comment (jimmygchen) [inline beacon_node/http_api/tests/broadcast_validation_tests.rs]:
```suggestion
    let fork_name = state.fork_name(&tester.harness.spec).unwrap();
```

**Review by jimmygchen — APPROVED** [review]
LGTM!
